### PR TITLE
Feature/admin completed functionality.

### PR DESCRIPTION
**Pull request by colegregory**
Finished basics of listing-detail. The forms in the listing-detail view are functional and any changes made in the number of credits / description of a listing can be saved and will be updated in the rest of YACS.

### CODE REVIEW

File: core/app/resources/listing_resource.rb [hunk 1]
```diff
@@ -6,11 +6,11 @@ class ListingResource < ApplicationResource
   has_many :sections
 
   attribute :longname, :string
```
Comment (Bad-Science) [inline core/app/resources/listing_resource.rb]: Can we make the longname editable as well?

File: web/src/app/listing-detail/component.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
+import { Component, OnInit } from '@angular/core';
+import {Location} from '@angular/common';
+import { Router, ActivatedRoute, Params } from '@angular/router';
+import { Term, Course, Listing } from 'yacs-api-client';
+import { YacsService } from '../services/yacs.service';
```
Comment (Bad-Science) [inline web/src/app/listing-detail/component.ts]: I don't think this is needed

File: web/src/app/listing-detail/component.html [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  <div id="nameField" class="form-group form-inline">
+    <label for="nameInput">Name: </label>
+    <input id="nameInput" class="form-control" style ="width:95% !important;"[(ngModel)]="course.name" placeholder="name" #newName>
+  </div>
+  </div> -->
+
+  <div class="form-row">
+    <div id="credField" class="form-group form-inline">
+      <label for="minCredInput">Min Credits: </label>
+      <input id="minCredInput" class="form-control" style="width: 70px !important" [(ngModel)]="listing.minCredits" placeholder="{{listing.minCredits}}" #newMinCred>
```
Comment (Bad-Science) [inline web/src/app/listing-detail/component.html]: Don't use inline styles

File: web/src/app/listing/component.html [hunk 4]
```diff
@@ -1,21 +1,24 @@
 <div class = "course" [class.fixed-height]="!showingMenu && !showingDescription" (mouseenter)="hovered=true" (mouseleave) = "hovered=false">
 	<div class="row flex-nowrap h-100 min-height">
-	  <div class="indicator-container" (click)="clickCourse()">
-	  	<div class="indicator h-100" [class.indicator-selected]="isCourseSelected()" [class.indicator-unselected]="!isCourseSelected()" [class.indicator-hover]="hovered && !isCourseSelected()" [class.indicator-selected-hover]="hovered && isCourseSelected()"></div>
-	  </div>
-	  <div class="w-100 h-100 course-content position-relative min-height">
-		  <table (click)="clickCourse()" class="w-100">
-		  	<tr>
-		    	<div class="course-name">{{listing.longname}} 
+		<div class="indicator-container" (click)="clickCourse()">
```
Comment (Bad-Science) [inline web/src/app/listing/component.html]: Looks like the spacing in this file was unintentionally modified, please fix.

File: web/src/app/listing/component.scss [hunk 5]
```diff
@@ -1,5 +1,10 @@
 @import "../../_vars.scss";
 
+a {
```
Comment (Bad-Science) [inline web/src/app/listing/component.scss]: inconsistent spacing

File: web/src/app/listing-detail/component.ts (hunk 2 above)
Comment (colegregory) [reply web/src/app/listing-detail/component.ts]: Do you mean for all the imports? I use Location, ActivatedRoute, Params, and Listing but I can get rid of the other ones.

Comment (Bad-Science) [reply web/src/app/listing-detail/component.ts]: Talking about `YacsService`

**Comment by colegregory** [comment]
Totally forgot to finish this.

**Comment by Bad-Science** [comment]
Heheh no problem @colegregory . Can you post some screenshots? :)

**Comment by colegregory** [comment]
Screenshots:
Screen Shot 2019-03-20 at 8 35 47 PM

Screen Shot 2019-03-20 at 8 36 10 PM

File: web/src/app/listing-detail/component.ts [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+      this.goBack();
+    }
+
+    goBack (): void {
+      this.location.back();
+    }
+
+    ngOnInit (): void {
+      this.activatedRoute.params.subscribe((params: Params) => {
+        this.id = +params['id'];
```
Comment (Bad-Science) [inline web/src/app/listing-detail/component.ts]: is this a typo?

Comment (colegregory) [reply web/src/app/listing-detail/component.ts]: yes

**Pull request closed without merging**